### ISSUE TITLE
Experiment Module: Update 'Plan' entity

### ISSUE THREAD

**Issue by aislinnpearson** [Rothamsted-Ecoinformatics/farm_rothamsted#358]
Update the Experiment module's 'Plan' with additional data fields. See https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/issues/298#issuecomment-1411784802

**Comment by paul121** [comment]
Hey @aislinnpearson it's not clear to me which of the fields we need to add/change/remove for the existing plans... I think we spoke about this once and you said to ignore that part of the spreadsheet. But now seeing this issue I want to double check. Just let me know!

**Comment by aislinnpearson** [comment]
@paul121 A note to link to the updated file:

https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/issues/298#issuecomment-1439898263

**Comment by paul121** [comment]
@aislinnpearson I think there is still some remaining work I have to do here. Lets talk about this tomorrow

**Comment by aislinnpearson** [comment]
Hey @paul121,

Here is the list of fixes as possible - sorry, it's a long one but I think what we have come up with works well. FYI I called the 'plan', 'variable' and 'plot' tabs 'pages' just to separate them out in the document from what we were previously calling tabs. Let me know if there is a more appropriate word. :)

**Plan**

**Description Tab**
- 'Experiment design' data field: add description
- Add data fields: Description, Plan Number, Start Date, End Date, Crop(s), Current Phase, Planting Year, Harvest Year, Cost Code(s), Amendments

![image](https://user-images.githubusercontent.com/16349781/222185608-b6080774-08be-4935-a015-5cd753c65ab6.png)

**Location and Assets Tab**
- Move location and assets data fields from General Tab
- If possible, it would be nice to specifically display the experiment boundary and sensor assets, but this is not essential
- It might be nice to include a note on this tab to say "Plot assets are listed under the Plots page' 
![image](https://user-images.githubusercontent.com/16349781/222187845-03755167-14b9-4bb4-a28c-a23acbdf5252.png)

**Research Permissions tab:**
- Add data fields for editing and email notifications. After thinking it through last night I went for the safest and most restrictive option here of specifying Researchers with specific validation criteria. My thinking is that this way we have a very comprehensive list of who is involved and who can edit the plans. Let me know if this is even possible though - my brain is struggling with this a bit! :)

![image](https://user-images.githubusercontent.com/16349781/222189886-1bca92c5-2572-47db-b98a-79fa67becfed.png)

**Restrictions tab**
- Add all data fields

![image](https://user-images.githubusercontent.com/16349781/222190303-9d12a0df-ab8f-46b2-a714-84bad7baa5d0.png)

**Management Tab**
- Add all data fields

Image 1:
![image](https://user-images.githubusercontent.com/16349781/222190677-43c11fda-4399-4887-b946-3facf9552bd5.png)

Image 2 (i.e continued list of data fields for management tab): 
![image](https://user-images.githubusercontent.com/16349781/222190956-7947ca7a-4d73-48b2-8bc9-f34c737f0241.png)

**Files Tab**
- Add all data fields to a files tab (only the URL question is new though)
![image](https://user-images.githubusercontent.com/16349781/222191498-f00457d7-acbe-4da0-8e69-4062288248f4.png)

**Status Tab**
- No changes :)

![image](https://user-images.githubusercontent.com/16349781/222191650-74728a53-98fc-448e-baa5-67b1f5beba72.png)

**Variables Page:**
- No required changes but I thought I'd ask if we can make the variables expandable, so you see a list of variables and each variable can be expanded to show the levels (but just a nice to have, not essential) 
![image](https://user-images.githubusercontent.com/16349781/222194660-2e3fe2ef-2f5c-4a95-8218-7dfa3ce3ffc9.png)

**Plots Page:**
- From a user perspective, I was thinking be safer if we specify that both the plot_number and plot_id are required in the GeoJSON attributes. We can still only validate against plot_number, but this will force the user to double check their plot_number matches the plot_id. What do you think?
- I also realised we need an option to select existing plots for experiments which span multiple years. Is it possible to add an asset selection widget that only searches plot attributes?

![image](https://user-images.githubusercontent.com/16349781/222196192-02b578c6-af11-47d7-a051-0d7f5c9fa5c7.png)

And that should be everything! Feel free to give me a shout with any questions. I've also kept my Friday afternoon free just in case it's easier to chat all this through.

Thanks so much as always @paul121

**Comment by paul121** [comment]
I've made all of these changes. There a few small things @aislinnpearson and I chatted about and have decided to wait or just do differently:
- We'll wait on permission + email fields
- Instead of quantities, cost codes will re-use the existing cost code field (it now allows multiple) and have an additional text area for describing cost code allocations
- Crops will use the existing crop taxonomy field

**Comment by paul121** [comment]
@aislinnpearson one thing that wasn't covered here is what to do with the existing "Trial design" tab... I've moved all of these fields to other tabs as it makes sense.

But two fields that are leftover are the RRES experiment category & object fields. I put these in the "Description" tab for now. If you'd like to change/move/remove these fields lets do so in a new issue on the next milestone

**Comment by paul121** [comment]
This is done!

**Comment by aislinnpearson** [comment]
Hey @paul121 

> But two fields that are leftover are the RRES experiment category & object fields. I put these in the "Description" tab for now. If you'd like to change/move/remove these fields lets do so in a new issue on the next milestone

We do have fields for this... I'll make a 'Migration' issue though I think so we can define it all there and then also sign the migration off as done.
